app/dao/fact_notification_status_dao.py

```diff
@@ -313,6 +319,86 @@ def fetch_notification_status_for_service_for_today_and_7_previous_days(service_
     ).all()
 
 
+def fetch_daily_notification_status_for_service(service_id, by_template=False):
+    """
+    Fetches daily notification stats for a service
+
+    Service limits reset at 12:00am UTC each night, so we need to fetch the data from 12:00 UTC to now
+    """
+
+    start = utc_midnight_n_days_ago(1)
+
+    stats_for_today = (
+        db.session.query(
+            Notification.notification_type.cast(db.Text),
+            Notification.status,
+            *([Notification.template_id] if by_template else []),
+            *([func.count().label("count")]),
+        )
+        .filter(
+            Notification.created_at >= start,
+            Notification.service_id == service_id,
+            Notification.key_type != KEY_TYPE_TEST,
+        )
+        .group_by(
+            Notification.notification_type,
+            *([Notification.template_id] if by_template else []),
+            Notification.status,
+        )
+    )
+
+    all_stats_table = stats_for_today.subquery()
+
+    query = db.session.query(
+        *(
+            [
+                Template.name.label("template_name"),
+                Template.is_precompiled_letter,
```

Comment:
I am proabaly missing this but why do we need the line is_precompiled_letter?

Reply:
This is existing code - not sure.